### ISSUE TITLE
Fix `selector-no-invalid` false negatives for modern pseudo-elements like `:is()`

### ISSUE THREAD

**Issue by MorevM** [stylelint/stylelint#9400]
### What is the problem you're trying to solve?

Continuing the discussion started in #8588 

```css
/* 
 * Syntactically correct, but never selects anything
 */
:is(::placeholder) {}
:where(input::placeholder) {}
.foo:has(::before) {}
:not(::before) {}
:nth-child(2n of ::before) {}
:host(::before) {}
::slotted(::before) {}

/* 
 * Syntactically correct, but never selects anything AND
 * invalidates the whole rule (`:has()` is not forgiving)
 */
.card:has(.item, ::before) {}

/* 
 * Slightly different, but related:
 * `:has()` cannot be nested by the spec.
 */
.foo:has(.bar:has(.baz)) {}
```

[Spec about nested `:has`](https://www.w3.org/TR/selectors-4/?utm_source=chatgpt.com#:~:text=The%20%3Ahas()%20pseudo%2Dclass%20cannot%20be%20nested%3B)

### What solution would you like to see?

Perhaps this should be an extension of [`selector-no-invalid`](https://stylelint.io/user-guide/rules/selector-no-invalid/), for nested `:has()` at least.

Perhaps we need a new rule - the selector is actually _valid_, it's just always no-op.
`:has()` is riskier - while `:is` and `:where` are forgiving selector lists, `:has()` is not, and it invalidates the entire branch.

We need to be careful with `:has()` - it theoretically [allows pseudo-elements based on a whitelist](https://www.w3.org/TR/selectors-4/#:~:text=Also%2C%20unless%20explicitly%20defined%20as%20a%20%3Ahas%2Dallowed%20pseudo%2Delement), but the list is currently empty.

I'd be happy to submit a PR once we've discussed the design of the change / new rule.

---

I personally think it would be better to handle nested `:has()` as an extension of `selector-no-invalid` - it's _actually_ invalid, and address everything else in a new rule.

Naming options:
- `selector-pseudo-element-no-no-op` (just kidding)
- `selector-pseudo-element-no-empty` (but it won't necessarily be empty, since it's forgiving)
- `selector-pseudo-element-no-unmatchable` sounds most reasonable, and we have agreed-upon #9183.

What do you think?

**Comment by ybiquitous** [comment]
@MorevM Thank you for raising the discussion about the incorrect use of `:is()`/`:where()`/`:has()`.

It seems we have to be careful about extending `selector-no-invalid` here:

- `selector-no-invalid`'s implementation might be more complicated.
- `selector-no-invalid` might conflict with other current or new rules.

I'd like to hear ideas from other maintainers.

**Comment by MorevM** [comment]
@ybiquitous 
Should I create a draft concept for a new rule without the `selector-no-invalid` extension? 
It doesn't seem like there's much interest here, but I'd like to do it anyway.

**Comment by jeddy3** [comment]
@MorevM Thank you for your eagerness to get stuck in, and for offering to create a draft concept.

There is interest; we tend to rest during the summer months, so it's quiet at the moment. There's a fair bit to unpack to get the various `selector-*-unmatchable|invalid` rules right, but I've a pretty good idea of how the pieces might fit together. I'll hopefully have time to share in the next few weeks so we can discuss the approach and, if we're all happy with it, roll it out with a few new rules that play well together.

Thanks for your patience.
